runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTranslationContext.java

```diff
@@ -84,6 +85,17 @@ public void setOutputDataStream(PValue value, DataStream<?> set) {
     }
   }
 
+  <T extends PValue> void setProducer(T value, PTransform<?, T> producer) {
+    if (!producers.containsKey(value)) {
```

Comment:
Should this be turned to `Precondition.checkState`?

Reply:
👍 makes sense